lib/active_shipping/carriers/ups.rb

```diff
@@ -626,7 +626,9 @@ def response_success?(document)
     end
 
     def response_message(document)
-      document.root.at_xpath('Response/Error/ErrorDescription | Response/ResponseStatusDescription').text
```

Comment:
Your code is an improvement for sure, but does it now actually return the proper description when it wasn't before?